tutorials/shaders/shader_reference/shading_language.rst

```diff
@@ -516,6 +516,18 @@ Keep in mind that, in modern GPUs, an infinite loop can exist and can freeze
 your application (including editor). Godot can't protect you from this, so be
 careful not to make this mistake!
 
+.. note::
+
+    Branching in shaders can be expensive, especially if every pixel/vertex in a
+    given frame evaluates to different branches. GPUs are massively parallel
+    processors; they're fastest when the data they work with is *predictable*.
+
+    "Static" branching (where every pixel/vertex evaluates to ``true`` or
+    ``false`` in a frame) is less expensive, but it can still have a noticeable
+    cost if used too often. Depending on context, consider using the
+    :ref:`doc_shader_preprocessor`'s ``#if`` directive to reduce the number of
+    branches that are compiled in the final shader.
+
```

Comment:
I'm not sure this note makes sense here. It also isn't quite accurate. I would just remove it altogether